Winey/Winey/Source/Network/Base/Encodable.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  Encodable.swift
+//  Winey
+//
+//  Created by 김인영 on 2023/07/17.
+//
+
+import Foundation
+
+extension Encodable {
+    
+    func asParameter() throws -> [String: Any] {
+        let data = try JSONEncoder().encode(self)
+        guard let dictionary = try JSONSerialization.jsonObject(with: data, options: .allowFragments)
+                as? [String: Any] else {
+            throw NSError()
+        }
+        return dictionary
+    }
```

Comment:
P2) 좋네용 👍🏻

요거 의견을 조오금 남겨보자면,

1. extension에서는 에러를 던지기보다 nil을 던지는 것이 더 나은 선택같다는 생각이 듭니당 저희가 정의한 정의가 아닌, NSError는 더 파악하기가 어려울 것 같아용 에러는 사용처에서 정의하고 던지면 좋을 것 같슴다~!
2. 이름에 "s"가 들어가야할 것 같아용 파라미터는 복수가 가능하니까요..? 다른 이름으로는 asDictionary도 추천함다. 

아래처럼 수정해도 좋을 거 같아요~

```suggestion
    var dictionary: [String: Any] {
        guard let data = try? JSONEncoder().encode(self) else { return [:] }
        let result = (try? JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed))
            .flatMap { $0 as? [String: Any] }
        
        return result ?? [:]
    }
```

Reply:
이 부분은 조금 더 공부해보고 코드 수정할게요!